addons/crm/controllers/main.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import werkzeug
+from werkzeug import url_encode
+
+from openerp.addons.mail.controllers.main import MailController
+from openerp import http
+from openerp.http import request
+
+
+class CrmController(http.Controller):
+
+    @http.route('/lead/case_mark_won', type='http', auth='user', methods=['POST'], csrf=False)
+    def crm_lead_case_mark_won(self, model, res_id):
+        if model not in request.env:
+            return MailController._redirect_to_messaging()
+        Model = request.env[model]
```

Comment:
model should be hard-coded to `crm.lead`

Method should only take lead id as param.